### PR TITLE
feat(log): add tencent cls

### PR DESCRIPTION
**Pull request by Germiniku**
#### Description (what this PR does / why we need it):

this pr is a new feature.Add tencent cls log server.

### CODE REVIEW

File: contrib/log/tencent/tencent.go [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+	if err != nil {
+		return nil, err
+	}
+	return &tencentLog{
+		producer: producerInst,
+		opts:     opts,
+	}, nil
+}
+
+// toString 任意类型转string
```
Comment (Windfarer) [inline contrib/log/tencent/tencent.go:115]: in english

File: contrib/log/tencent/tencent.go [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+	return &options{}
+}
+
+func WithEndpoint(endpoint string) Option {
+	return func(cls *options) {
+		cls.endpoint = endpoint
+	}
+}
+
+func WithTopicId(topicId string) Option {
```
Comment (tonybase) [inline contrib/log/tencent/tencent.go:46]: ID

File: contrib/log/tencent/tencent.go [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+	logInst := &cls.Log{
+		Time:     proto.Int64(time.Now().Unix()),
+		Contents: contents,
+	}
+	a.producer.SendLog(a.opts.topicId, logInst, nil)
+	return nil
+}
+
+func NewTencentLog(options ...Option) (Log, error) {
```
Comment (tonybase) [inline contrib/log/tencent/tencent.go:96]: Tencent.NewLogger

File: contrib/log/tencent/tencent.go [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+	"encoding/json"
+	"strconv"
+	"time"
+
+	klog "github.com/go-kratos/kratos/v2/log"
+	cls "github.com/tencentcloud/tencentcloud-cls-sdk-go"
+	"google.golang.org/protobuf/proto"
+)
+
+type Log interface {
```
Comment (tonybase) [inline contrib/log/tencent/tencent.go:13]: Logger

File: contrib/log/tencent/tencent.go [hunk 5]
```diff
@@ -0,0 +1,155 @@
+package tencent
+
+import (
+	"encoding/json"
+	"strconv"
+	"time"
+
+	klog "github.com/go-kratos/kratos/v2/log"
```
Comment (tonybase) [inline contrib/log/tencent/tencent.go:8]: log

**Review by tonybase — DISMISSED** [review]

**Comment by Germiniku** [comment]
Now i fix these problems.

File: contrib/log/tencent/tencent.go [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+	case uint32:
+		key = strconv.Itoa(int(v))
+	case int64:
+		key = strconv.FormatInt(v, 10)
+	case uint64:
+		key = strconv.FormatUint(v, 10)
+	case string:
+		key = v
+	case []byte:
+		key = string(v)
```
Comment (tonybase) [inline contrib/log/tencent/tencent.go:150]: case fmt.Stringer:

Comment (Germiniku) [reply contrib/log/tencent/tencent.go]: Now i have resolved this.

**Review by tonybase — APPROVED** [review]

**Pull request merged**